iceoryx_binding_c/source/c_wait_set.cpp

```diff
@@ -316,3 +317,47 @@ void iox_ws_detach_client_state(iox_ws_t const self, iox_client_t const client,
 
     self->detachState(*client, c2cpp::clientState(clientState));
 }
+
+iox_WaitSetResult iox_ws_attach_service_discovery_event(const iox_ws_t self,
+                                                        const iox_service_discovery_t serviceDiscovery,
+                                                        const ENUM iox_ServiceDiscoveryEvent serviceDiscoveryEvent,
+                                                        const uint64_t eventId,
+                                                        void (*callback)(iox_service_discovery_t))
+{
+    iox::cxx::Expects(self != nullptr);
```

Comment:
Actually, we could just call here:
```
return iox_ws_attach_service_discovery_event_with_context_data(self, serviceDiscovery, serviceDiscoveryEvent, eventId, callback, nullptr);
```

and remove all the code.

Reply:
Unfortunately not, because the signature of the callbacks don't match.